compiler/plc_xml/src/model/pou.rs

```diff
@@ -1,30 +1,42 @@
-use std::{collections::HashMap, str::FromStr};
+use std::{borrow::Cow, collections::HashMap, str::FromStr};
 
+use plc_diagnostics::diagnostics::Diagnostic;
 use quick_xml::events::Event;
 
 use crate::{error::Error, extensions::GetOrErr, reader::PeekableReader, xml_parser::Parseable};
 
 use super::{action::Action, body::Body, interface::Interface};
 
 #[derive(Debug, Default)]
-pub(crate) struct Pou {
-    pub name: String,
+pub(crate) struct Pou<'xml> {
+    pub name: Cow<'xml, str>,
```

Comment:
Can you explain why the Cow is useful in all these structs?
I ask myself, since you anyway provide a lifetime, is there a difference to storing a `&'xml str`? this cow somehow suggests to me that at one point you want to mutate this field?

Reply:
in the case below (line 22) we get a `String` out of the attributes hashmap, so didnt this already perform the clone that you try to avoid here? maybe I'm misunderstanding

Reply:
_theoratically_ the Cow will not be cloning until it is modified. Not sure how efficient it is here. @mhasel probably has some other thoughts on why he used it.

Reply:
> _theoratically_ the Cow will not be cloning until it is modified. Not sure how efficient it is here. @mhasel probably has some other thoughts on why he used it.

thats what the documentation says. but again:
1. I dont see where we would mutate the nodes
2. I dont see where we would really share these strings?
we use it like this: `name: Cow::from(attributes.get_or_err("name")?)` and the get_or_error returns a String by ownership
```rust
    fn get_or_err(&self, key: &str) -> Result<String, Error> {
        self.get(key).map(|it| it.to_owned()).ok_or(Error::MissingAttribute(key.to_string()))
    }
```
so do you think anything will be shared here?

Reply:
In my initial solution, I was attempting to clone the node-index to have an immutable lookup while mutably iterating through the original index. I've experimented with Cow<str> to see if I could make the clone as cheap as possible, however, this turned out to be a moot point, since when I tested the Cow variants, all of them were of the Cow::Owned kind.
At that point, I could have changed back all the fields to String, but I figured we don't lose anything from using Cow<str>. So yes, your point is valid: these aren't technically needed. Since there is no downside to using these over Strings here, though, I decided to just keep them as they are.
I have thought about trying these in the AST aswell, and it might be worth a try to see if we save some time due to fewer allocations. Then there is also Arc<str>, similar idea.

Reply:
ok thx for clearification - we can keep them from my perspective